lib/config/configuration.go

```diff
@@ -486,11 +504,15 @@ func applyLogConfig(loggerConfig Log, logger *log.Logger) error {
 			return trace.Wrap(err)
 		}
 
-		logger.SetFormatter(formatter)
+		setFormatter(formatter)
+		cfg.Console = ioutil.Discard     // disable console printing
```

Comment:
Looks like we set `Console = nil` most of the time. I see only two places where we set it to `io.Discard`. We should try to be consistent.

Reply:
> It looks like the solution to this issue boils down to setting service.Config.Console=nil when the logging configuration is set to stdout or stderr

when the format is set to json, it doesn't matter where the logs are going

> I think we should just configure and use the root logger

This seem to contradict this old comment
`// DELETE this when global logger instance is no longer in use.`
Which makes me think the intent was to phase out the StandardLogger.

In any case I don't mind spending more time on this if someone think we should broaden the scope of the linked issue to unifying logging across the codebase.

Reply:
> Looks like we set Console = nil most of the time. I see only two places where we set it to io.Discard. We should try to be consistent.

Just looked into this. It looks like it being set to `ioutil.Discard` anyway, so might as well keep it.
https://github.com/gravitational/teleport/blob/502323590933ad4145f40f001070833556cc0ecf/lib/service/service.go#L3850-L3852

Reply:
> when the format is set to json, it doesn't matter where the logs are going

I don't understand. Isn’t the issue that `utils.Consolef` writes two different formats to same output? This doesn't really seem like it’s about json formatting. We just happened to catch the issue because we were outputting both json and non-json, which broke a json parser. 

The issue seems to be that `Consolef` assumes the log is being written to something other than stdout. I don’t think there would be a reason for `Consolef` to duplicate a message to the same output. LMK if I’m missing something!

> This seem to contradict this old comment

Yeah, we’ve been discussing ways to improve how we configure logging. The desire to not use the standard logger created a situation where different parts of the system used their own logger. Something as simple as disabling logging for tests became a challenge because we don’t have a single root logger. 

We need a root logger so we can configure logging for the entire process. Many places already use the standard logger because it is a sensible solution, so the standard logger is a reasonable choice for a root logger. 

> In any case I don't mind spending more time on this if someone think we should broaden the scope

I think it is always a good idea to fix things as we find them! Especially if the solution produces less complexity. Using the standard logger should reduce this PR to a handful of lines changed. Wouldn't it be better to configure a single logger rather than two?

> Just looked into this. It looks like its being set to ioutil.Discard anyway, so might as well keep it.

Up to you (although that location should be changed to `io.Discard`). I'd rather see consistency one way or the other to avoid confusion. All of our tests set it to `nil` because `utils.Consolef` checks for `nil`. The two locations that set it to discard are doing so needlessly because the only place `Console` is used is as a parameter to `utils.Consolef`.

```
integration/app_integration_test.go|1009 col 9| rcConf.Console = nil
integration/app_integration_test.go|1025 col 9| lcConf.Console = nil
integration/app_integration_test.go|1463 col 10| raConf.Console = nil
integration/app_integration_test.go|1527 col 10| laConf.Console = nil
integration/helpers.go|397 col 8| tconf.Console = console
integration/helpers.go|983 col 8| tconf.Console = nil
integration/integration_test.go|5644 col 6| cfg.Console = nil
integration/kube_integration_test.go|1191 col 8| tconf.Console = nil
integration/proxy_helpers_test.go|168 col 9| tconf.Console = nil
config/configuration.go|1752 col 7| cfg.Console = ioutil.Discard
service/cfg.go|287 col 20| Console:     cfg.Console,
service/cfg.go|1111 col 6| cfg.Console = os.Stdout
service/service.go|3858 col 9| if cfg.Console == nil {
service/service.go|3859 col 7| cfg.Console = ioutil.Discard
```

Reply:
I have difficulty to understand the purpose of `cfg.Console` usage. 

The comments says 
```
	// Console writer to speak to a user
	Console io.Writer
```

Where by default the cfg.Console is set to `cfg.Console = os.Stdout` 

For `text` logs produced by `Consolef` will duplicated for log entry : 
```
2022-03-10T13:33:06+01:00 INFO [KUBERNETE] Kubernetes service 9.0.0-beta.1:v6.1.6-0-gefab0f364-dirty is starting on [::]:6354. utils/cli.go:273
[KUBERNETES]   Kubernetes service 9.0.0-beta.1:v6.1.6-0-gefab0f364-dirty is starting on [::]:6354.
```


and before that change for json formatting: 
```
{"caller":"utils/cli.go:273","component":"kubernetes:1","level":"info","message":"Kubernetes service 9.0.0-beta.1:v6.1.6-0-gefab0f364-dirty is starting on [::]:6354.","timestamp":"2022-03-10T13:32:12+01:00"}
[KUBERNETES]   Kubernetes service 9.0.0-beta.1:v6.1.6-0-gefab0f364-dirty is starting on [::]:6354.
```

So if we want to remove the `[KUBERNETES]   Kubernetes servic` line for json formatting we can also remove this line for text and get rid of `cfg.Consolef`

Reply:
Sorry for the late reply, got held up working on other more urgent issues.

@jimbishopp perfect. I'll just make the switch and see where we go from there
@smallinsky yes I also agree that's the usage and purpose is a bit confusing and it needs refactor for it to work the way it was intended to. ie log and console should not be able to write to the same location...